modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/plugin/ArtifactPluginDescriptorFactory.java

```diff
@@ -77,50 +109,38 @@ public ArtifactPluginDescriptor create(File pluginFolder) throws ArtifactDescrip
     }
 
+    loadUrlsToClassLoaderModelBuilder(pluginFolder, classLoaderModelBuilder);
+
+    descriptor.setClassLoaderModel(classLoaderModelBuilder.build());
+    descriptor.setBundleDescriptor(createDefaultPluginBundleDescriptor(descriptor.getName()));
+    return descriptor;
+  }
+
+  /**
+   * Given a {@code pluginFolder} as a starting point, it will add a /classes and every JAR file {@link URL}s into the
+   * {@link ClassLoaderModelBuilder}.
+   *
+   * @param pluginFolder plugin's location
+   * @param classLoaderModelBuilder builder of the {@link ClassLoaderModel}
+   */
+  private void loadUrlsToClassLoaderModelBuilder(File pluginFolder, ClassLoaderModelBuilder classLoaderModelBuilder) {
     try {
       classLoaderModelBuilder.containing(new File(pluginFolder, "classes").toURI().toURL());
       final File libDir = new File(pluginFolder, "lib");
       if (libDir.exists()) {
         final File[] jars = libDir.listFiles((FilenameFilter) new SuffixFileFilter(".jar"));
-        for (int i = 0; i < jars.length; i++) {
-          classLoaderModelBuilder.containing(jars[i].toURI().toURL());
+        for (File jar : jars) {
+          classLoaderModelBuilder.containing(jar.toURI().toURL());
         }
       }
     } catch (MalformedURLException e) {
       throw new ArtifactDescriptorCreateException("Failed to create plugin descriptor " + pluginFolder);
     }
-
-    descriptor.setClassLoaderModel(classLoaderModelBuilder.build());
-    descriptor.setExtensionModelDescriptorProperty(lookupJsonDescriptor(pluginFolder));
-    descriptor.setBundleDescriptor(createDefaultPluginBundleDescriptor(descriptor.getName()));
-
-    return descriptor;
-  }
-
-  private LoaderDescriber lookupJsonDescriptor(File pluginFolder) {
-    LoaderDescriber loaderDescriber = null;
-    final File mulePluginJsonFile = new File(pluginFolder, META_INF + separator + MULE_PLUGIN_JSON);
-    if (mulePluginJsonFile.exists()) {
-
-      MulePluginModel mulePluginModel = getMulePluginJsonDescriber(mulePluginJsonFile);
-      Optional<MulePluginLoaderDescriptor> extensionModelDescriptor = mulePluginModel.getExtensionModelLoaderDescriptor();
-      if (extensionModelDescriptor.isPresent()) {
-        loaderDescriber = new LoaderDescriber(extensionModelDescriptor.get().getId());
-        loaderDescriber.addAttributes(extensionModelDescriptor.get().getAttributes());
-      }
-    }
-    return loaderDescriber;
-  }
-
-  private MulePluginModel getMulePluginJsonDescriber(File jsonFile) {
-    try (InputStream stream = new FileInputStream(jsonFile)) {
-      return new MulePluginModelJsonSerializer().deserialize(IOUtils.toString(stream));
-    } catch (IOException e) {
-      throw new IllegalArgumentException(format("Could not read extension describer on plugin '%s'", jsonFile.getAbsolutePath()),
-                                         e);
-    }
   }
 
+  /**
+   * // TODO(fernandezlautaro): MULE-11092 once implemented, drop this.
```

Comment:
Don't add a TODO for each thing that must be removed, detecting what has to be cleaned up is part of the work once the loadFromPluginProperties method is removed